benchmark/src/grpc_helpers.rs

```diff
@@ -162,22 +170,97 @@ fn handle_account_state_response(
     }
 }
 
-/// Query a bunch of accounts' sequence numbers and status from an validator.
-/// If operation for one account fails, during either querying or handling, return immediately.
-/// When everything is fine, the return vector, whose item is (sequence_number, account_status),
-/// is ensured to be in consistent order with the given account_address vector.
+/// Request a bunch of accounts' states, including sequence numbers and status from validator.
+/// Ignore any failure, during either requesting or processing, and continue for next account.
+/// Return the mapping from address to (sequence number, account status) tuple
+/// for all successfully requested accounts.
 pub fn get_account_states(
     client: &AdmissionControlClient,
-    account_address: &[AccountAddress],
-) -> Result<Vec<(u64, AccountStatus)>> {
-    let future_resps: Result<Vec<_>> = account_address
+    addresses: &[AccountAddress],
+) -> HashMap<AccountAddress, (u64, AccountStatus)> {
+    let futures: Vec<_> = addresses
         .iter()
-        .map(|address| get_account_state_async(client, address))
+        .filter_map(|address| match get_account_state_async(client, *address) {
+            Ok(future) => Some(future),
+            Err(e) => {
+                error!("Failed to send account request: {:?}", e);
+                None
+            }
+        })
         .collect();
-    future_resps?
-        .into_iter()
-        .map(handle_account_state_response)
-        .collect()
+    let future_stream = stream::futures_unordered(futures);
```

Comment:
Reply to Sonia:

Now after changing the return type in `get_account_state_async`, we don't need to use `buffer_unordered`.

To answer your origin question: in order to put stream into buffer, the item in the stream not only need to be a Future, but also implement IntoFuture, say an Option or Result.